packages/react/src/drafts/UnderlinePanels/UnderlinePanels.tsx

```diff
@@ -0,0 +1,193 @@
+import React, {Children, isValidElement, cloneElement, useState, useRef, type FC, type PropsWithChildren} from 'react'
+import {TabContainerElement} from '@github/tab-container-element'
+import {createComponent} from '../../utils/create-component'
+import {
+  StyledUnderlineItemList,
+  StyledUnderlineWrapper,
+  UnderlineItem,
+  type UnderlineItemProps,
+} from '../../internal/components/UnderlineTabbedInterface'
+import Box, {type BoxProps} from '../../Box'
+import {useId} from '../../hooks'
+import {invariant} from '../../utils/invariant'
+import type {IconProps} from '@primer/octicons-react'
+import {merge, type BetterSystemStyleObject, type SxProp} from '../../sx'
+import {defaultSxProp} from '../../utils/defaultSxProp'
+import {useResizeObserver, type ResizeObserverEntry} from '../../hooks/useResizeObserver'
+import useIsomorphicLayoutEffect from '../../utils/useIsomorphicLayoutEffect'
+
+export type UnderlinePanelsProps = {
+  /**
+   * Tabs (UnderlinePanels.Tab) and panels (UnderlinePanels.Panel) to render
+   */
+  children: React.ReactNode
+  /**
+   * Accessible name for the tab list
+   */
+  'aria-label'?: React.AriaAttributes['aria-label']
+  /**
+   * ID of the element containing the name for the tab list
+   */
+  'aria-labelledby'?: React.AriaAttributes['aria-labelledby']
+  /**
+   * Custom string to use when generating the IDs of tabs and `aria-labelledby` for the panels
+   */
+  id?: string
+  /**
+   * Loading state for all counters. It displays loading animation for individual counters until all are resolved. It is needed to prevent multiple layout shift.
+   */
+  loadingCounters?: boolean
+} & SxProp
+
+export type TabProps = PropsWithChildren<{
+  /**
+   * Whether this is the selected tab
+   */
+  'aria-selected'?: boolean
```

Comment:
I wonder if we should also add `aria-controls` attribute to each tab and associate the values to their respective panels. (based on [APG Tab Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/)) but also [there is not enough support](https://a11ysupport.io/tech/aria/aria-controls_attribute) for `aria-controls` 🤔 What are your thoughts?

Reply:
We don't use `aria-controls` for other tabs, so I wouldn't introduce it here. For example, we don't attempt setting `aria-controls` in [@github/tab-container-element](https://github.com/github/tab-container-element)